app/routes/companyInterest/CompanyInterestEditRoute.js

```diff
@@ -11,13 +11,13 @@ import CompanyInterestPage, {
   OTHER_TYPES
 } from './components/CompanyInterestPage';
 import { selectCompanyInterestById } from 'app/reducers/companyInterest';
-import { selectCompanySemesters } from 'app/reducers/companySemesters';
+import { selectCompanySemestersForInterestform } from 'app/reducers/companySemesters';
 import { push } from 'react-router-redux';
 import prepare from 'app/utils/prepare';
 
 const loadCompanyInterests = (props, dispatch) => {
   const { companyInterestId } = props.params;
-  return dispatch(fetchSemesters()).then(() =>
+  return dispatch(fetchSemesters({ companyInterest: 'True' })).then(() =>
```

Comment:
Hmm, I dont like the 'True' value – maybe better to pass a boolean and fix the query in the action? Add better flow typing would be nice as well, because it is easy to make errors with these strings.

Reply:
Agree,

This was just a quickfix by having the same behavior in the edit and create form. Currently the edit form lists all semesters in the db. Aka. 19. I also added a custom selector to only select active semesters, so this isn't that necessary. This action should just be something like `fetchSemestersForCompanyIntrests`, that is a small wrapper around the `fetchSemesters` action.

Reply:
![screenshot from 2017-10-26 13-03-53](https://user-images.githubusercontent.com/1467188/32049676-2ea7edb8-ba4e-11e7-9d3e-80de6147acf4.png)